VERIFY in test_ttl_delete: (ui32)result->num_columns() == columnNames.size();

```
E   VERIFY failed (2025-03-21T00:34:05.250882+0300): verification=(ui32)result->num_columns() == columnNames.size();fline=process_columns.cpp:106;schema=ts: timestamp[us];required=_yql_plan_step,_yql_tx_id,_yql_write_id;
E   ydb/library/actors/core/log.cpp:754
E   ~TVerifyFormattedRecordWriter(): requirement false failed
E   0. /-S/util/system/yassert.cpp:83: NPrivate::InternalPanicImpl(int, char const*, char const*, int, int, int, TBasicStringBuf<char, std::__y1::char_traits<char>>, char const*, unsigned long) @ 0x9FC5F35
E   1. /-S/util/system/yassert.cpp:55: NPrivate::Panic(NPrivate::TStaticBuf const&, int, char const*, char const*, char const*, ...) @ 0x9FBECAC
E   2. /-S/ydb/library/actors/core/log.cpp:754: NActors::TVerifyFormattedRecordWriter::~TVerifyFormattedRecordWriter() @ 0xAD39893
E   3. /-S/ydb/core/formats/arrow/process_columns.cpp:106: std::__y1::shared_ptr<arrow::Table> NKikimr::NArrow::(anonymous namespace)::ExtractImpl<arrow::Table, TBasicString<char, std::__y1::char_traits<char>>>(NKikimr::NArrow::TColumnOperator::EAbsentFieldPolicy const&, std::__y1::shared_ptr<arrow::Table> const&, std::__y1::vector<TBasicString<char, std::__y1::char_traits<char>>, std::__y1::allocator<TBasicString<char, std::__y1::char_traits<char>>>> const&) @ 0xD0E4D53
E   4. /-S/ydb/core/formats/arrow/process_columns.cpp:188: NKikimr::NArrow::TColumnOperator::Extract(std::__y1::shared_ptr<arrow::Table> const&, std::__y1::vector<TBasicString<char, std::__y1::char_traits<char>>, std::__y1::allocator<TBasicString<char, std::__y1::char_traits<char>>>> const&) @ 0xD0E4D53
E   5. /-S/ydb/core/tx/columnshard/engines/reader/plain_reader/iterator/merge.cpp:107: NKikimr::NOlap::NReader::NPlain::TStartMergeTask::DoExecuteImpl() @ 0x1419457E
E   6. /-S/ydb/core/tx/columnshard/engines/reader/common/conveyor_task.cpp:8: NKikimr::NOlap::NReader::IDataTasksProcessor::ITask::DoExecute(std::__y1::shared_ptr<NKikimr::NConveyor::ITask> const&) @ 0x141E1920
E   7. /-S/ydb/core/tx/conveyor/usage/abstract.cpp:16: NKikimr::NConveyor::ITask::Execute(std::__y1::shared_ptr<NKikimr::NConveyor::TTaskSignals>, std::__y1::shared_ptr<NKikimr::NConveyor::ITask> const&) @ 0x1414F06F
E   8. /-S/ydb/core/tx/conveyor/service/worker.cpp:11: NKikimr::NConveyor::TWorker::ExecuteTask(std::__y1::vector<NKikimr::NConveyor::TWorkerTask, std::__y1::allocator<NKikimr::NConveyor::TWorkerTask>>&&) @ 0x1A9B03ED
E   9. /-S/ydb/core/tx/conveyor/service/worker.cpp:47: NKikimr::NConveyor::TWorker::HandleMain(TAutoPtr<NActors::TEventHandle<NKikimr::NConveyor::TEvInternal::TEvNewTask>, TDelete>&) @ 0x1A9B187B
E   10. /-S/ydb/library/actors/core/actor.cpp:280: NActors::IActor::Receive(TAutoPtr<NActors::IEventHandle, TDelete>&) @ 0xACF39D4
E   11. /-S/ydb/library/actors/core/executor_thread.cpp:269: NActors::TExecutorThread::Execute(NActors::TMailbox*, bool) @ 0xAD2B480
E   12. /-S/ydb/library/actors/core/executor_thread.cpp:460: NActors::TExecutorThread::ProcessExecutorPool()::$_0::operator()(NActors::TMailbox*, bool) const @ 0xAD2EF81
E   13. /-S/ydb/library/actors/core/executor_thread.cpp:512: NActors::TExecutorThread::ProcessExecutorPool() @ 0xAD2EBD0
E   14. /-S/ydb/library/actors/core/executor_thread.cpp:538: NActors::TExecutorThread::ThreadProc() @ 0xAD2F74A
E   15. /-S/util/system/thread.cpp:244: (anonymous namespace)::TPosixThread::ThreadProxy(void*) @ 0x9FC6D8C
E   16. ??:0: ?? @ 0x7F44E4E1E608
E   17. ??:0: ?? @ 0x7F44E4D43352
```

To reproduce:
```
ya make --build=relwithdebinfo -tA -F '*test_ttl_delete' -P ydb/tests/olap
```

Revision: main 10580ef92d154905b27fae13b961c3ff56bd7791

--- a/.github/config/muted_ya.txt
+++ b/.github/config/muted_ya.txt
@@ -249,7 +249,6 @@ ydb/tests/olap/ttl_tiering data_migration_when_alter_ttl.py.TestDataMigrationWhe
 ydb/tests/olap/ttl_tiering sole chunk chunk
 ydb/tests/olap/ttl_tiering ttl_delete_s3.py.TestDeleteS3Ttl.test_data_unchanged_after_ttl_change
 ydb/tests/olap/ttl_tiering ttl_delete_s3.py.TestDeleteS3Ttl.test_delete_s3_tiering
-ydb/tests/olap/ttl_tiering ttl_delete_s3.py.TestDeleteS3Ttl.test_ttl_delete
 ydb/tests/olap/ttl_tiering unstable_connection.py.TestUnstableConnection.test
 ydb/tests/postgres_integrations/go-libpq [docker_wrapper_test.py] chunk chunk
 ydb/tests/postgres_integrations/go-libpq docker_wrapper_test.py.test_pg_generated[Test64BitErrorChecking]

--- a/ydb/core/kqp/ut/olap/kqp_olap_ut.cpp
+++ b/ydb/core/kqp/ut/olap/kqp_olap_ut.cpp
@@ -3305,5 +3305,33 @@ Y_UNIT_TEST_SUITE(KqpOlap) {
             CompareYson("[[[\"aa\"];10u;\"n1\"];[[\"cc\"];30u;\"n3\"]]", FormatResultSetYson(result.GetResultSet(0)));
         }
     }
+
+    Y_UNIT_TEST(CountWithPredicate) {
+        NKikimrConfig::TAppConfig appConfig;
+        appConfig.MutableTableServiceConfig()->SetEnableOlapSink(true);
+        auto runnerSettings = TKikimrSettings().SetAppConfig(appConfig).SetWithSampleTables(true);
+
+        TTestHelper testHelper(runnerSettings);
+        auto client = testHelper.GetKikimr().GetQueryClient();
+
+        TVector<TTestHelper::TColumnSchema> schema = {
+            TTestHelper::TColumnSchema().SetName("time").SetType(NScheme::NTypeIds::Timestamp).SetNullable(false),
+            TTestHelper::TColumnSchema().SetName("class").SetType(NScheme::NTypeIds::Utf8).SetNullable(false),
+        };
+
+        TTestHelper::TColumnTable testTable;
+        testTable.SetName("/Root/ColumnTableTest").SetPrimaryKey({ "time", "class" }).SetSchema(schema);
+        testHelper.CreateTable(testTable);
+
+        auto ts = TInstant::Now();
+        {
+            TTestHelper::TUpdatesBuilder tableInserter(testTable.GetArrowSchema(schema));
+            tableInserter.AddRow().Add(100).Add("test");
+            tableInserter.AddRow().Add(ts.MicroSeconds() * 2).Add("test");
+            testHelper.BulkUpsert(testTable, tableInserter);
+        }
+
+        testHelper.ReadData("SELECT COUNT(*) FROM `/Root/ColumnTableTest` WHERE time > CurrentUtcTimestamp()", "[[1u]]");
+    }
 }
 }

--- a/ydb/core/tx/columnshard/engines/reader/plain_reader/iterator/context.cpp
+++ b/ydb/core/tx/columnshard/engines/reader/plain_reader/iterator/context.cpp
@@ -76,6 +76,7 @@ std::shared_ptr<TFetchingScript> TSpecialReadContext::DoGetColumnsFetchingPlan(c
             NCommon::TFetchingScriptBuilder acc(*this);
             acc.SetBranchName("FAKE");
             acc.AddStep(std::make_shared<TBuildFakeSpec>());
+            acc.AddStep(std::make_shared<NCommon::TBuildStageResultStep>());
             return std::move(acc).Build();
         }
     }
@@ -157,6 +158,8 @@ std::shared_ptr<TFetchingScript> TSpecialReadContext::BuildColumnsFetchingPlan(c
         if (needSnapshots || GetFFColumns()->Cross(*GetSpecColumns())) {
             acc.AddAssembleStep(*GetSpecColumns(), "SPEC", EStageFeaturesIndexes::Filter, false);
             acc.AddStep(std::make_shared<TSnapshotFilter>());
+        } else if (GetProgramInputColumns()->Cross(*GetSpecColumns())) {
+            acc.AddStep(std::make_shared<TBuildFakeSpec>());
         }
         acc.AddFetchingStep(*GetFFColumns(), EStageFeaturesIndexes::Fetching);
         acc.AddAssembleStep(*GetFFColumns(), "LAST", EStageFeaturesIndexes::Fetching, !exclusiveSource);

--- a/ydb/core/tx/columnshard/engines/reader/plain_reader/iterator/fetching.cpp
+++ b/ydb/core/tx/columnshard/engines/reader/plain_reader/iterator/fetching.cpp
@@ -98,7 +98,6 @@ TConclusion<bool> TBuildFakeSpec::DoExecuteInplace(const std::shared_ptr<IDataSo
             std::make_shared<NArrow::NAccessor::TTrivialArray>(
                 NArrow::TThreadSimpleArraysCache::GetConst(f->type(), NArrow::DefaultScalar(f->type()), source->GetRecordsCount())), true);
     }
-    source->BuildStageResult(source);
     return true;
 }
 

